internal/elasticsearch/rbac.go

```diff
@@ -13,9 +13,10 @@ import (
 func (er *ElasticsearchRequest) CreateOrUpdateRBAC() error {
 	dpl := er.cluster
 
-	// elasticsearch RBAC
-	elasticsearchRole := rbac.NewClusterRole(
```

Comment:
IIRC this is the only place where we need to create Cluster-scope RBAC resources. If yes, then I suggest to remove also the manifest functions for `NewClusterRole` and `NewClusterRoleBinding`.